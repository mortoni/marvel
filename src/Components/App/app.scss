.App {
  text-align: center;
  height: 100vh;
  font-size: 1.2rem;
}

.comments {
    font-size: 1rem;
}

.char-container {
    overflow: scroll;
    height: 100%;
    padding: 1.5rem;
}
