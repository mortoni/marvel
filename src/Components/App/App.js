import React from 'react';
import './app.scss';
import Header from '../../Components/Header';
import Character from '../../Components/Character';
import ModalContent from '../../Components/ModalContent';
import Modal from 'react-responsive-modal';
import { connect } from 'react-redux';
import { openModal,
         closeModal,
         searchTerm,
         updateChars } from '../../Actions';
import PropTypes from 'prop-types';

export const App = (props) =>  {
    const { openModal,
            closeModal,
            modal,
            search,
            searchTerm,
            updateChars } = props;

    const handleInputChange = (query) => {
        searchTerm(query);
        if (query.length + 1 > 2) {
            updateChars(query);
        }
    }

    const content = () => {
        return modal.isOpen ? <ModalContent { ...modal.char }/> : null;
    }

    return (
        <div className="App">
            <Header
                handleInputChange={ handleInputChange }
                isSearching={ search.isSearching }/>

            <div className="char-container">
                <div className="row">
                    { search.chars.map(char => (
                        <Character
                            key={ char.id }
                            { ...char }
                            onOpenModal={ openModal }
                        />
                    )) }
                </div>

                <Modal open={ modal.isOpen } onClose={ closeModal } center>
                  { content() }
                </Modal>

            </div>
        </div>
    );
}

App.propTypes = {
    openModal: PropTypes.func.isRequired,
    closeModal: PropTypes.func.isRequired,
    modal: PropTypes.object.isRequired,
    search: PropTypes.object.isRequired,
    searchTerm: PropTypes.func.isRequired,
    updateChars: PropTypes.func.isRequired,
};

function mapStateToProps({ modal, search }) {
    return {
        modal,
        search,
    };
}

function mapDispatchToProps(dispatch) {
    return {
        openModal: (char) => dispatch(openModal(char)),
        closeModal: () => dispatch(closeModal()),
        searchTerm: (query) => dispatch(searchTerm(query)),
        updateChars: (query) => dispatch(updateChars(query)),
    };
}

export default connect(mapStateToProps, mapDispatchToProps)(App);
