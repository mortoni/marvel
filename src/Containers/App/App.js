import React, { Component } from 'react';
import './App.scss';
import { getCharacters } from '../../Api';
import Header from '../../Components/Header';

class App extends Component {
    componentDidMount() {
        getCharacters().then(data => {
            const test = data;
        });
    }

    handleInputChange = (query) => {

    }

    render() {
        return (
            <div className="App">
                <Header handleInputChange={ this.handleInputChange }/>
            </div>
        );
    }
}

export default App;
