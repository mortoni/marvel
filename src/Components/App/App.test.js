import React from 'react';
import { App } from './App';
import enzymeAdapterReact16 from 'enzyme-adapter-react-16';
import Enzyme, { render, shallow } from 'enzyme';

Enzyme.configure({ adapter: new enzymeAdapterReact16() });

describe('Testing App component', () => {
    const props = {
        closeModal: () => {},
        search: {
            chars: []
        },
        modal: {
            isOpen: false
        },
        openModal: () => {},
        searchTerm: () => {},
        updateChars: () => {},
        cleanChars: () => {},
    };

    it('renders correctly', () => {
        const wrapper = shallow(<App {...props}/>);
        expect(wrapper).toBeDefined();
    });

    it('renders as expected', () => {
        const wrapper = shallow(<App {...props}/>);
        expect(wrapper).toMatchSnapshot();
    });
});
