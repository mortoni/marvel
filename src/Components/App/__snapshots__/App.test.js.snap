// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing App component renders as expected 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <App
    closeModal={[Function]}
    modal={
      Object {
        "isOpen": false,
      }
    }
    openModal={[Function]}
    search={
      Object {
        "chars": Array [],
      }
    }
    searchTerm={[Function]}
    updateChars={[Function]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Header
          handleInputChange={[Function]}
        />,
        <div
          className="char-container"
        >
          <div
            className="row"
          />
          <Modal
            animationDuration={500}
            blockScroll={true}
            center={true}
            classNames={Object {}}
            classes={
              Object {
                "closeButton": "styles_closeButton__20ID4",
                "closeIcon": "styles_closeIcon__1QwbI",
                "modal": "styles_modal__gNwvD",
                "overlay": "styles_overlay__CLSq-",
                "overlayCenter": "styles_overlayCenter__YHoO7",
                "transitionEnter": "styles_transitionEnter__3j_-a",
                "transitionEnterActive": "styles_transitionEnterActive___eQs7",
                "transitionExit": "styles_transitionExit__1KmEf",
                "transitionExitActive": "styles_transitionExitActive__1nQXw",
              }
            }
            closeIconSize={28}
            closeIconSvgPath={
              <path
                d="M28.5 9.62L26.38 7.5 18 15.88 9.62 7.5 7.5 9.62 15.88 18 7.5 26.38l2.12 2.12L18 20.12l8.38 8.38 2.12-2.12L20.12 18z"
              />
            }
            closeOnEsc={true}
            closeOnOverlayClick={true}
            focusTrapOptions={Object {}}
            focusTrapped={false}
            onClose={[Function]}
            onEntered={null}
            onEscKeyDown={null}
            onExited={null}
            onOverlayClick={null}
            open={false}
            showCloseIcon={true}
            styles={Object {}}
          />
        </div>,
      ],
      "className": "App",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "handleInputChange": [Function],
          "isSearching": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="row"
            />,
            <Modal
              animationDuration={500}
              blockScroll={true}
              center={true}
              classNames={Object {}}
              classes={
                Object {
                  "closeButton": "styles_closeButton__20ID4",
                  "closeIcon": "styles_closeIcon__1QwbI",
                  "modal": "styles_modal__gNwvD",
                  "overlay": "styles_overlay__CLSq-",
                  "overlayCenter": "styles_overlayCenter__YHoO7",
                  "transitionEnter": "styles_transitionEnter__3j_-a",
                  "transitionEnterActive": "styles_transitionEnterActive___eQs7",
                  "transitionExit": "styles_transitionExit__1KmEf",
                  "transitionExitActive": "styles_transitionExitActive__1nQXw",
                }
              }
              closeIconSize={28}
              closeIconSvgPath={
                <path
                  d="M28.5 9.62L26.38 7.5 18 15.88 9.62 7.5 7.5 9.62 15.88 18 7.5 26.38l2.12 2.12L18 20.12l8.38 8.38 2.12-2.12L20.12 18z"
                />
              }
              closeOnEsc={true}
              closeOnOverlayClick={true}
              focusTrapOptions={Object {}}
              focusTrapped={false}
              onClose={[Function]}
              onEntered={null}
              onEscKeyDown={null}
              onExited={null}
              onOverlayClick={null}
              open={false}
              showCloseIcon={true}
              styles={Object {}}
            />,
          ],
          "className": "char-container",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [],
              "className": "row",
            },
            "ref": null,
            "rendered": Array [],
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "animationDuration": 500,
              "blockScroll": true,
              "center": true,
              "children": null,
              "classNames": Object {},
              "classes": Object {
                "closeButton": "styles_closeButton__20ID4",
                "closeIcon": "styles_closeIcon__1QwbI",
                "modal": "styles_modal__gNwvD",
                "overlay": "styles_overlay__CLSq-",
                "overlayCenter": "styles_overlayCenter__YHoO7",
                "transitionEnter": "styles_transitionEnter__3j_-a",
                "transitionEnterActive": "styles_transitionEnterActive___eQs7",
                "transitionExit": "styles_transitionExit__1KmEf",
                "transitionExitActive": "styles_transitionExitActive__1nQXw",
              },
              "closeIconSize": 28,
              "closeIconSvgPath": <path
                d="M28.5 9.62L26.38 7.5 18 15.88 9.62 7.5 7.5 9.62 15.88 18 7.5 26.38l2.12 2.12L18 20.12l8.38 8.38 2.12-2.12L20.12 18z"
              />,
              "closeOnEsc": true,
              "closeOnOverlayClick": true,
              "focusTrapOptions": Object {},
              "focusTrapped": false,
              "onClose": [Function],
              "onEntered": null,
              "onEscKeyDown": null,
              "onExited": null,
              "onOverlayClick": null,
              "open": false,
              "showCloseIcon": true,
              "styles": Object {},
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Header
            handleInputChange={[Function]}
          />,
          <div
            className="char-container"
          >
            <div
              className="row"
            />
            <Modal
              animationDuration={500}
              blockScroll={true}
              center={true}
              classNames={Object {}}
              classes={
                Object {
                  "closeButton": "styles_closeButton__20ID4",
                  "closeIcon": "styles_closeIcon__1QwbI",
                  "modal": "styles_modal__gNwvD",
                  "overlay": "styles_overlay__CLSq-",
                  "overlayCenter": "styles_overlayCenter__YHoO7",
                  "transitionEnter": "styles_transitionEnter__3j_-a",
                  "transitionEnterActive": "styles_transitionEnterActive___eQs7",
                  "transitionExit": "styles_transitionExit__1KmEf",
                  "transitionExitActive": "styles_transitionExitActive__1nQXw",
                }
              }
              closeIconSize={28}
              closeIconSvgPath={
                <path
                  d="M28.5 9.62L26.38 7.5 18 15.88 9.62 7.5 7.5 9.62 15.88 18 7.5 26.38l2.12 2.12L18 20.12l8.38 8.38 2.12-2.12L20.12 18z"
                />
              }
              closeOnEsc={true}
              closeOnOverlayClick={true}
              focusTrapOptions={Object {}}
              focusTrapped={false}
              onClose={[Function]}
              onEntered={null}
              onEscKeyDown={null}
              onExited={null}
              onOverlayClick={null}
              open={false}
              showCloseIcon={true}
              styles={Object {}}
            />
          </div>,
        ],
        "className": "App",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "handleInputChange": [Function],
            "isSearching": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                className="row"
              />,
              <Modal
                animationDuration={500}
                blockScroll={true}
                center={true}
                classNames={Object {}}
                classes={
                  Object {
                    "closeButton": "styles_closeButton__20ID4",
                    "closeIcon": "styles_closeIcon__1QwbI",
                    "modal": "styles_modal__gNwvD",
                    "overlay": "styles_overlay__CLSq-",
                    "overlayCenter": "styles_overlayCenter__YHoO7",
                    "transitionEnter": "styles_transitionEnter__3j_-a",
                    "transitionEnterActive": "styles_transitionEnterActive___eQs7",
                    "transitionExit": "styles_transitionExit__1KmEf",
                    "transitionExitActive": "styles_transitionExitActive__1nQXw",
                  }
                }
                closeIconSize={28}
                closeIconSvgPath={
                  <path
                    d="M28.5 9.62L26.38 7.5 18 15.88 9.62 7.5 7.5 9.62 15.88 18 7.5 26.38l2.12 2.12L18 20.12l8.38 8.38 2.12-2.12L20.12 18z"
                  />
                }
                closeOnEsc={true}
                closeOnOverlayClick={true}
                focusTrapOptions={Object {}}
                focusTrapped={false}
                onClose={[Function]}
                onEntered={null}
                onEscKeyDown={null}
                onExited={null}
                onOverlayClick={null}
                open={false}
                showCloseIcon={true}
                styles={Object {}}
              />,
            ],
            "className": "char-container",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [],
                "className": "row",
              },
              "ref": null,
              "rendered": Array [],
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "animationDuration": 500,
                "blockScroll": true,
                "center": true,
                "children": null,
                "classNames": Object {},
                "classes": Object {
                  "closeButton": "styles_closeButton__20ID4",
                  "closeIcon": "styles_closeIcon__1QwbI",
                  "modal": "styles_modal__gNwvD",
                  "overlay": "styles_overlay__CLSq-",
                  "overlayCenter": "styles_overlayCenter__YHoO7",
                  "transitionEnter": "styles_transitionEnter__3j_-a",
                  "transitionEnterActive": "styles_transitionEnterActive___eQs7",
                  "transitionExit": "styles_transitionExit__1KmEf",
                  "transitionExitActive": "styles_transitionExitActive__1nQXw",
                },
                "closeIconSize": 28,
                "closeIconSvgPath": <path
                  d="M28.5 9.62L26.38 7.5 18 15.88 9.62 7.5 7.5 9.62 15.88 18 7.5 26.38l2.12 2.12L18 20.12l8.38 8.38 2.12-2.12L20.12 18z"
                />,
                "closeOnEsc": true,
                "closeOnOverlayClick": true,
                "focusTrapOptions": Object {},
                "focusTrapped": false,
                "onClose": [Function],
                "onEntered": null,
                "onEscKeyDown": null,
                "onExited": null,
                "onOverlayClick": null,
                "open": false,
                "showCloseIcon": true,
                "styles": Object {},
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
